src/cargo/ops/cargo_compile.rs

```diff
@@ -361,6 +363,16 @@ fn generate_targets<'a>(pkg: &'a Package,
             Ok(targets)
         }
     };
+
+    targets.map(|x| x.iter().filter_map(|&(t, p)| {
```

Comment:
Also, instead of having `targets` be of the type `CargoResult<Vec<_>>` it's probably best to have it just be `Vec<_>` (e.g. remove all the `Ok` above) and just wrap it in `Ok` after the filtering is done.

Ah and as a final point I think this can use `retain` instead of `iter` + `filter` + `collect`